conformance/conformance_test.cc

```diff
@@ -299,6 +314,7 @@ void ConformanceTestSuite::RunValidInputTest(
   RunTest(test_name, request, &response);
 
   TestAllTypes test_message;
+  TestAllTypesProto2 test_message_proto2;
```

Comment:
Too much code duplication. There should be a `Message*` that points to the correct message. Then message->ParseFromString() will do the right thing.